sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -0,0 +1,341 @@
+      else {
+        setAlign(RequiredAlign);
+        MShadowCopy = allocateHostMem();
+        MUserPtr = MShadowCopy;
+        std::memcpy(MUserPtr, HostPtr.get(), MSizeInBytes);
+      }
+    }
+  }
+
+  template <class InputIterator>
+  void handleHostData(InputIterator First, InputIterator Last,
+                      const size_t RequiredAlign) {
+    MHostPtrReadOnly = iterator_to_const_type_t<InputIterator>::value;
+    setAlign(RequiredAlign);
+    if (useHostPtr())
+      throw invalid_parameter_error(
+          "Buffer constructor from a pair of iterator values does not support "
+          "use_host_ptr property.");
+
+    setAlign(RequiredAlign);
+    MShadowCopy = allocateHostMem();
+    MUserPtr = MShadowCopy;
+
+    // We need to cast MUserPtr to pointer to the iteration type to get correct
+    // offset in std::copy when it will increment destination pointer.
+    using IteratorValueType = iterator_value_type_t<InputIterator>;
+    using IteratorNonConstValueType = remove_const_t<IteratorValueType>;
+    using IteratorPointerToNonConstValueType =
+        add_pointer_t<IteratorNonConstValueType>;
+    std::copy(First, Last,
+              static_cast<IteratorPointerToNonConstValueType>(MUserPtr));
+  }
+
+  template <typename T = AllocatorT>
+  EnableIfNonDefaultAllocator<T> setAlign(size_t RequiredAlign) {
+    // Do nothing in case of user's allocator.
+  }
+
+  template <typename T = AllocatorT>
+  EnableIfDefaultAllocator<T> setAlign(size_t RequiredAlign) {
+    MAllocator.setAlignment(std::max<size_t>(RequiredAlign, 64));
+  }
+
+protected:
+  // Allocator used for allocation memory on host.
+  AllocatorT MAllocator;
+  // Properties passed by user.
+  property_list MProps;
+  // Event passed by user to interoperability constructor.
+  // Should wait on this event before start working with such memory object.
+  EventImplPtr MInteropEvent;
+  // Context passed by user to interoperability constructor.
+  ContextImplPtr MInteropContext;
+  // OpenCL's memory object handle passed by user to interoperability
+  // constructor.
+  cl_mem MInteropMemObject;
+  // Indicates whether memory object is created using interoperability
+  // constructor or not.
+  bool MOpenCLInterop;
+  // Indicates if user provided pointer is read only.
+  bool MHostPtrReadOnly;
+  // Indicates if memory object should write memory to the host on destruction.
+  bool MNeedWriteBack;
+  // Size of memory.
+  size_t MSizeInBytes;
+  // User's pointer passed to constructor.
+  void *MUserPtr;
+  // Copy of memory passed by user to constructor.
+  void *MShadowCopy;
+  // Function which update host with final data on memory object destruction.
+  std::function<void(void)> MUploadDataFunctor;
+  // Field which holds user's shared_ptr in case of memory object is created
+  // using constructor with shared_ptr.
+  shared_ptr_class<const void> MSharedPtrStorage;
+};
+
+} // namespace detail
+} // namespace sycl
+} // namespace cl
```

Comment:
Missing end-of-line...

Reply:
Not sure how to fix it, I don't see a problem. Could you clarify?